thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/views/AnalysisView.js

```diff
@@ -39,18 +40,26 @@ AnalysisView.prototype = {
       }
     });
 
-    analysisMetricSelect.on("change", function(e) {
-      var selectedElement = $(e.currentTarget);
-      var selectedData = selectedElement.select2("data");
-      var metricId = selectedData.map(function (e) {return e.id})[0];
-      var metricAlias = selectedData.map(function (e) {return e.text})[0];
-      var metricName = selectedData.map(function (e) {return e.name})[0];
-      self.viewParams['metric'] = {id: metricId, alias: metricAlias, allowClear:true, name:metricName};
+    analysisMetricSelect.on("change", (e) => {
+      const selectedElement = $(e.currentTarget);
+      const selectedData = selectedElement.select2("data");
+      let metricId;
+      let metricAlias;
+      let metricName;
+
+      if (selectedData.length) {
+        metricId = selectedData.map(function (e) {return e.id})[0];
```

Comment:
It looks iterating the entire selectedData array but it only needs the first one.
Let's pick the first index data and then spread the values.
```
const { id, text, name } = selectedData[0];
metricId = id;
metricAlias = text;
metricName = name;
```